File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
import java.util.Queue;
import java.util.LinkedList;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        Queue <Integer> que = new LinkedList<>();
        
        int cnt = 0;
        
        for (int i =0; i<n; i++){
            if (visited[i]){
                continue;
            }
            else{
                que.offer(i);
                answer++;
                
                while(!que.isEmpty()){
                    int current = que.poll();
                    visited[current] = true;
                     for (int j =0; j<n; j++){
                        if (visited[j]){
                            continue;
                        }
                        else{
                            if(computers[current][j] == 1 && current != j){
                                que.offer(j);
                                visited[j] = true;
                            }
                            else{
                                continue;
                            }
                        }
                        
                    }
                }

            }
        }
        
        
        
        return answer;
    }
    
    
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    def dfs(node, visited):
        visited[node] = True
        for i in range(n):
            if computers[node][i] == 1 and not visited[i]:
                dfs(i, visited)

    visited = [False] * n
    answer = 0

    for i in range(n):
        if not visited[i]:  # 방문하지 않은 컴퓨터 발견 시
            dfs(i, visited)
            answer += 1

    return answer